core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryParser.java

```diff
@@ -103,7 +104,7 @@ public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOExcept
                     unit = DistanceUnit.fromString(parser.text());
                 } else if ("distance_type".equals(currentFieldName) || "distanceType".equals(currentFieldName)) {
                     geoDistance = GeoDistance.fromString(parser.text());
-                } else if (currentFieldName.endsWith(GeoPointFieldMapper.Names.LAT_SUFFIX)) {
+                } else if (currentFieldName.endsWith(GeoPointFieldMapper.Names.LAT_SUFFIX)) { // TODO IDE complains currentFieldName could be null here - add null check?
```

Comment:
I don't think we should initialise to empty string. Why not add an assert here to check for null? This should never be null here as that would mean we got a value before a field name.